File: БР1.2/Соболев Артём/labs/lab2/FitnessAPI/common-lib/src/main/java/io/github/artsobol/common/config/properties/security/JwtProperties.java
package io.github.artsobol.common.config.properties.security;

import org.springframework.boot.context.properties.ConfigurationProperties;

import java.time.Duration;

@ConfigurationProperties(prefix = "app.security.jwt")
public record JwtProperties(
        String secret,
        Duration accessTokenExpiration
) {
}

File: БР1.2/Цатинян Артём/labs/lab1/src/main/kotlin/ru/itmo/restaurantbooking/auth/config/JwtProperties.kt
package ru.itmo.restaurantbooking.auth.config

import org.springframework.boot.context.properties.ConfigurationProperties
import java.nio.charset.StandardCharsets

@ConfigurationProperties(prefix = "app.jwt")
data class JwtProperties(
    val issuer: String,
    val accessTokenTtlMinutes: Long,
    val secret: String
) {
    init {
        require(secret.toByteArray(StandardCharsets.UTF_8).size >= 32) {
            "JWT secret must contain at least 32 bytes for HS256"
        }
    }
}
